src/deploy/utils.py

```
@@ -356,6 +356,30 @@ def read_script_content(script_path: str, base_path: Optional[Path] = None) -> s
+    size = len(payload.encode("utf-8"))
+    if size > PIPEDREAM_CODE_BYTE_LIMIT:
+        # Fail here, loudly, with the number. Pipedream's own failure for this is
+        # a MySQL error surfaced only inside the step's Test panel, after the
+        # editor has already accepted the paste — by then the workflow is stuck
+        # in DEPLOY PENDING and nothing in the deploy output says why.
+        raise ValueError(
+            f"{script_path}: deploy payload is {size:,} bytes after stripping "
+            f"comments and docstrings, over Pipedream's {PIPEDREAM_CODE_BYTE_LIMIT:,}-byte "
+            f"step-code limit by {size - PIPEDREAM_CODE_BYTE_LIMIT:,}. "
+            f"(source is {len(source.encode('utf-8')):,} bytes)"
+        )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)